src/TSTransformer/util/spreadDestruction/spreadDestructArray.ts

```diff
@@ -0,0 +1,19 @@
+import luau from "@roblox-ts/luau-ast";
+import { TransformState } from "TSTransformer/classes/TransformState";
+
+export function spreadDestructArray(state: TransformState, parentId: luau.AnyIdentifier, index: number) {
+	const restId = state.pushToVar(luau.array(), "rest");
+	state.prereq(
+		luau.create(luau.SyntaxKind.CallStatement, {
+			expression: luau.call(luau.globals.table.move, [
```

Comment:
We still need to create a seperate table to clone into?

Reply:
Yes, otherwise `table.move` just moves things inside the existing table. The way you've done it now (in [9f36b69](https://github.com/roblox-ts/roblox-ts/pull/2761/commits/9f36b69fccc44d7401f6fb8492020ae1e5d56729)) looks good.